src/components/button.tsx

```diff
@@ -1,20 +1,28 @@
+"use client";
+
 import Link from "next/link";
-import React, { Children } from "react";
+import React, { useRef } from "react";
 
 interface Props {
   url?: string;
   clickFn?: () => void;
   children: React.ReactNode;
   btnclass: string;
+  type?: "button" | "submit" | "reset";
 }
 
-const Button = ({ url, clickFn, children, btnclass }: Props) => {
+const Button = ({ url, clickFn, children, btnclass, type }: Props) => {
+  let buttonRef = useRef<HTMLButtonElement>(null);
+  let linkRef = useRef<HTMLAnchorElement>(null);
```

Comment:
since you only need one ref at a time (either for the button or the link), you can use a single ref and dynamically assign it based on the condition

Reply:
i intially tried to do ref = useref<htmlbutton| htmlanchort>(null) but the complied wouldnt acceptit. using chat gpt gave me this as a suggestion

Reply:
that's correct so if the compile didn't accept it your issue was probably on my next comment line. 
got to be careful with gpt as sometimes it will give you a lazy or false answer